### PR TITLE
feat: implement OPCODE Origin

### LINKED ISSUE #32: Opcode: ORIGIN

### PR DESCRIPTION
**Pull request by Danikim01**
closes #32 
This PR implements the opcode Origin

### CODE REVIEW

File: src/syscall.rs [hunk 1]
```diff
@@ -31,6 +31,35 @@ pub struct U256 {
...
+}
+
+impl U256 {
+    pub fn from_be_bytes(bytes: [u8; 32]) -> Self {
+        let hi = u128::from_be_bytes(bytes[0..16].try_into().unwrap());
+        let lo = u128::from_be_bytes(bytes[16..32].try_into().unwrap());
+        U256 { hi, lo }
+    }
+
+    pub fn store_h160_in_u256(&mut self, value: &ethereum_types::H160) {
```
Comment (JulianVentura) [inline src/syscall.rs:48]: I think the name `copy_from_address` would be more clear for this function. So, I would also change the `value` type to `Address`. And remember you should import the primitive types from `src/primitives.rs`

File: src/syscall.rs [hunk 2]
```diff
@@ -31,6 +31,35 @@ pub struct U256 {
     pub hi: u128,
 }
 
+impl std::fmt::Display for U256 {
```
Comment (JulianVentura) [inline src/syscall.rs:34]: Do we need to implement `Display`?

File: src/syscall.rs [hunk 3]
```diff
@@ -191,6 +220,11 @@ impl<'c> SyscallContext<'c> {
         self.env.tx.data.len() as u32
     }
 
+    pub extern "C" fn get_origin(&self, address: &mut U256) {
+        let aux: &ethereum_types::H160 = &self.env.tx.caller;
```
Comment (JulianVentura) [inline src/syscall.rs:224]: I'm not sure, but I think that the `tx.caller` corresponds to the `CALLER` opcode. Here you are implementing `ORIGIN`.

File: tests/evm.rs [hunk 4]
```diff
@@ -114,6 +114,36 @@ fn fibonacci_example() {
     assert_eq!(number, 55_u32.into());
 }
 
+#[test]
+fn test_opcode_origin() {
```
Comment (JulianVentura) [inline tests/evm.rs:118]: Can we add a test (or change this one) to match the example given in [the EVM Codes playground](https://www.evm.codes/playground?unit=Wei&codeType=Mnemonic&code=%27ORIGIN%27_)? I have done something similar for `CALLER` opcode. Maybe that's useful to you.

**Review by azteca1998 — APPROVED** [review]

**Review by azteca1998 — APPROVED** [review]

**Pull request merged**